src/locks/locks/blocklocks/BlockLock.ts

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Location, Material, Rotation } from 'org.bukkit';
+import { Block, BlockFace } from 'org.bukkit.block';
+import { Directional } from 'org.bukkit.block.data';
+import { ItemFrame, Player } from 'org.bukkit.entity';
+import { LockCustomItem, LockDataType, LockItem } from '../lock-items';
+
+export interface BlockLockProps {
+  itemFrame: ItemFrame;
+  lockData: LockDataType;
+  lockCustomItem: LockCustomItem;
+  block: Block;
+}
+
+export abstract class BlockLock {
+  protected itemFrame: ItemFrame;
+  protected lockData: LockDataType;
+  protected lockCustomItem: LockCustomItem;
+  protected block: Block;
+
+  public static materials: Material[] = [];
+
+  protected constructor(props: BlockLockProps) {
+    this.itemFrame = props.itemFrame;
+    this.lockData = props.lockData;
+    this.lockCustomItem = props.lockCustomItem;
+    this.block = props.block;
+  }
+
+  public get location(): Location {
+    return this.itemFrame.location;
+  }
+
+  public isLocked() {
```

Comment:
This could be JS getter; additionally, the === true check is probably not needed, because isLocked is (or at least, should be?) required member.